### ISSUE TITLE
Unexpected error message when accessing protected properties from derived classes

### ISSUE THREAD

**Issue by jimitndiaye** [Microsoft/TypeScript#10637]
**TypeScript Version:**  2.0.0

I'm not sure if this is a problem with typescript@2.0.0 or with tslint@3.15.1 but I'm getting error messages about the following in VSCode (retargeted to use the installed version of typescript above):
**Code**

```ts
export class Shape {
 protected size: number;
}
export class Square extends Shape {
}
export class Circle extends  Shape {
 constructor(square: Square) {
 // The line below gives me an error something like "Property 'size' is protected and only accessible through an instance of class 'Circle'"
// Even if I cast the square instance to Shape I still get the error, even though Circle derives from Shape too.
  console.log(square.size); 
 }
}
```

**Expected behavior:**
I would expect to be able to access protected properties of class Shape above in a derived class (Circle)
**Actual behavior:** 
I get an error message.

**Comment by RyanCavanaugh** [comment]
Same reasoning as in C#: http://stackoverflow.com/questions/1904782/whats-the-real-reason-for-preventing-protected-member-access-through-a-base-sib

**Comment by jimitndiaye** [comment]
Ah. Thanks.
